apis/v1beta1/config.go

```diff
@@ -75,6 +87,37 @@ func (c *AnyConfig) MarshalJSON() ([]byte, error) {
 	return json.Marshal(c.Object)
 }
 
+// Pipeline is a struct of component type to a list of component IDs.
+type Pipeline struct {
+	Exporters  []string `json:"exporters" yaml:"exporters"`
+	Processors []string `json:"processors" yaml:"processors"`
+	Receivers  []string `json:"receivers" yaml:"receivers"`
+}
+
+// GetEnabledComponents constructs a list of enabled components by component type.
+func (c *Config) GetEnabledComponents() map[ComponentType]map[string]interface{} {
+	toReturn := map[ComponentType]map[string]interface{}{
+		ComponentTypeReceiver:  {},
+		ComponentTypeProcessor: {},
+		ComponentTypeExporter:  {},
+	}
```

Comment:
There's a slight ambiguity here. We return connectors as both exporters and receivers, which is technically correct as far as their role in the pipeline goes, but I wonder if it won't cause confusion in the future. If I were a caller of this function, I'd expect to only get actual receivers in the Receiver map.

Reply:
iirc this is technically the same as we would do today. After we do #2915 we could probably separate these out more effectively. Right now this is only used for parsing logic (generating RBAC and Ports) but obv in the future this should probably contain enabled extensions and connectors.